suitcase/csv/__init__.py

```diff
@@ -53,28 +71,34 @@ def export(gen, filepath):
             elif name == 'stop':
                 meta['stop'] = doc
             elif name == 'descriptor':
-                stream_name = doc.get('name')
+                stream_name = doc['name']
                 meta['descriptors'][stream_name].append(doc)
                 filepath_ = (f"{filepath}_{stream_name}_"
                              f"{next(desc_counters[doc['uid']])}.csv")
                 files[doc['uid']] = open(filepath_, 'w+')
-            elif name == 'event':
-                row = ', '.join(map(str, (doc['time'], *doc['data'].values())))
-                f = files[doc['descriptor']]
-                f.write(f'{row}\n')
+            elif name == 'event' or name == 'bulk_event':
+                # NOTE: this also works for bulk_events, it relies on
+                # doc['data']['some_key'] and doc['time'] both being either a
+                # value or a list of the same length.
+                if name == 'event':
+                    index = [doc['time']]
+                else:
+                    index = doc['time']
+                event_data = pandas.DataFrame(doc['data'], index=index)
+
+                event_data.to_csv(files[doc['descriptor']], mode='a',
+                                  header=doc['descriptor'] not in has_header,
+                                  index_label='time', **kwargs)
+                has_header.add(doc['descriptor'])
+
     finally:
         for f in files.values():
             f.close()
     with open(f"{filepath}_meta.json", 'w') as f:
         json.dump(meta, f)
     return (f.name,) + tuple(f.name for f in files.values())
-#
-# def ingest(...):
-#     ...
-#
-#
-# def reflect(...):
-#     ...
-#
-#
-# handlers = []
+
+
+class IllegalArgumentError(ValueError):
+    # This class is just included to make it easier to deal with errors later
```

Comment:
Is this used for kwargs, like checking header or mode those default values ?

Reply:
yes it is used to ensure that the kwargs that the 3 kwargs that are passed to the pandas.DataFrame.to_csv(...) call do not conflict with the 3 kwargs that are required to be hard set in this code.